### Урок 6. Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.###

1) Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy.
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [76]:
import numpy as np
import scipy.stats as stats
from math import sqrt

In [62]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])

In [63]:
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [65]:
cov = np.mean(zp*ks) - np.mean(zp)*np.mean(ks)
cov

9157.839999999997

In [67]:
np.cov(zp,ks, ddof =0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [68]:
np.std(zp)

59.115480206118605

In [69]:
np.std(ks)

174.55340157098058

In [70]:
cor = 9157.84/(59.115480206118605*174.55340157098058)
cor

0.8874900920739162

In [71]:
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

2) Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [72]:
a = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [77]:
np.mean(a) + stats.t.ppf(0.025, 10-1)*(np.std(a,ddof=1)/sqrt(10))

110.55608365158724

In [78]:
np.mean(a) - stats.t.ppf(0.025, 10-1)*(np.std(a,ddof=1)/sqrt(10))

125.64391634841274

Математическое ожидание находится в интервале **[110.556, 125.643]**, с надежностью 0,95

3) Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [80]:
174.2 + stats.norm.ppf(0.025)*(5/sqrt(27))

172.3140237765397

In [81]:
174.2 - stats.norm.ppf(0.025)*(5/sqrt(27))

176.08597622346028

Математическое ожидание находится в интервале **[172.314, 176.085]**, с надежностью 0,95